imap_processing/swe/l2/swe_l2.py

```diff
@@ -169,3 +174,44 @@ def calculate_flux(l1b_dataset: xr.Dataset) -> npt.NDArray:
         * phase_space_density_ds["phase_space_density"].data
     )
     return flux
+
+
+def swe_l2(l1b_dataset: xr.Dataset, data_version: str) -> xr.Dataset:
+    """
+    Will process data to L2.
+
+    Parameters
+    ----------
+    l1b_dataset : xarray.Dataset
+        The L1B dataset to process.
+    data_version : str
+        Version of the data product being created.
+
+    Returns
+    -------
+    data : xarray.Dataset
+        Processed data to L2.
+    """
+    # calculate flux
+    flux = calculate_flux(l1b_dataset)
+
+    # Calculate spin phase using SWE sci_step_acq_time_sec calculated in l1b.
+    # L1B dataset stores it by (epoch, energy, angle, cem).
+    data_acq_time = l1b_dataset["sci_step_acq_time_sec"].data.flatten()
+
+    # calculate spin phase
+    spin_phase = get_spacecraft_spin_phase(
```

Comment:
Does SWE want the spacecraft spin phase, or instrument spin phase. In case they want instrument spin phase, see this function: https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/dev/imap_processing/spice/geometry.py#L237

Reply:
SWE algorithm document says
>As part of Level 2 processing, the spin sectors are converted to spin angles of the spacecraft by combining SWE timing information with knowledge of the spacecraft spin from spacecraft ephemeris data.

This says spacecraft spin angles. What do you think? And I can double check with Ruth when I meet her after new year.

Reply:
And I asked her if I can use formula we discussed a while ago for that `get_spacecraft_spin_phase` function and she said, yes. But I wanted to double check this week but she couldn't meet.

Reply:
for sure. Will do!